aten/src/ATen/core/ivalue_inl.h

```diff
@@ -252,31 +252,43 @@ struct C10_EXPORT ivalue::Future final : c10::intrusive_ptr_target {
     completed_ = true;
```

Comment:
The `void wait()` API also need to be updated with the predicate put into the `cond_var.wait(lock, predicate)`? 
Now it might miss the notification and hangs there.

Reply:
The wait() condvar code should be correct as-is (note the outer while loop); the recipe there is about as textbook as things come.

(And for that matter, every place where completed_ is set to true, we have the lock and then notify_all)

The main issue that this change addresses is deadlocks arising from holding onto the lock while making callbacks)

Reply:
Thanks, now I see the subtlety here.